flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/polling/PollingRecordPublisher.java

```diff
@@ -85,12 +88,8 @@
 
 		Preconditions.checkArgument(expiredIteratorBackoffMillis >= 0);
 		Preconditions.checkArgument(maxNumberOfRecordsPerFetch > 0);
-	}
 
-	@Override
-	public void initialize(StartingPosition startingPosition) throws InterruptedException {
-		nextStartingPosition = startingPosition;
```

Comment:
With the `intialize` method now removed, could you also remove the preconditions check on line 101?
`nextStartingPosition` should now never be null.
Also, can `nextStartingPosition` now be a final field?

Reply:
Ah yes I missed that Precondition. I will remove that check. 
`nextStartingPosition` cannot be `final` as it is updated in `run` to recover in case of `ExpiredIteratorException`.